### PR TITLE
feat: Variant inventory quantity in GET requests

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**
- Add `inventory_quantity` to `GET /admin/products/:id/variants`
- Refactor the utils for `inventory_quantity` computation to be exposed to both `/store` and `/admin` routes

### CODE REVIEW

**Comment by olivermrbl** [comment]
@fPolic, in #7694, the available quantity that we show will live top-level on variants after merging this PR. So you can do:
`variant.inventory_quantity`

File: packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  if (!variantIds.length) {
+    return
+  }
+
+  const variantInventoriesMap = await getVariantInventoryItems({
+    req,
+    variantIds,
+  })
+
+  for (const variant of variants || []) {
```
Comment (fPolic) [inline packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts]: nit: if `variants` here can be undefined then we also need check on L:85

**Review by fPolic — APPROVED** [review]

File: packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+export async function getVariantInventoryItems({
+  req,
+  variantIds,
+  additionalFilters = {},
+  asMap = true,
+}) {
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+
+  const linkQuery = remoteQueryObjectFromString({
+    entryPoint: "product_variant_inventory_item",
```
Comment (adrien2p) [inline packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts]: suggestion: for the link query, you can use `service: LINKS.ProductVariantInventoryItem`

File: packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+export const wrapVariantsWithInventoryQuantity = async (
+  req: MedusaRequest,
+  variants: {
+    id: string
+    inventory_quantity?: number
+    manage_inventory?: boolean
+  }[]
+) => {
+  const variantIds = variants.map((variant) => variant.id).flat(1)
```
Comment (adrien2p) [inline packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts:87]:
```suggestion
  variants ??= [] // you might need to type it
  const variantIds = variants.map((variant) => variant.id).flat(1)
```

File: packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts (hunk 2 above)
Comment (adrien2p) [reply packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts]: This approach should be the preferred one a bit like resolving from ModuleRegistrationName etc wdyt?

**Review by adrien2p — APPROVED** [review]

File: packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts (hunk 2 above)
Comment (olivermrbl) [reply packages/medusa/src/api/utils/middlewares/products/variant-inventory-quantity.ts]: Yes good push, let me add that in

**Pull request merged**